src/pint/models/troposphere_delay.py

```diff
@@ -0,0 +1,368 @@
+        for lInd in range(len(self.LAT)):
+            if absLat >= self.LAT[lInd - 1]:
+                return lInd
+        # else this is an invalid latitude... huh?
+        raise ValueError("Invaid latitude: %s must be between -90 and 90 degrees" % lat)
+
+    def mapping_function(self, alt, lat, H, mjd):
+        """this implements the niell mapping function for hydrostatic
+        """
+
+        yearFraction = self._get_year_fraction_fast(mjd, lat)
+        """
+        according to Niell, the way to use latitude interpolation is to interpolate
+        between the nearest definite latitude coefficient functions.  So that means
+        I need to calcualte the function, then I can go back and interpolate between the results.
+        I figure the easiest way to do this will be to calculate the function on the entire array
+        """
+
+        # first I need to find the nearest latitude neighbors
+        latIndex = self._find_latitude_index(lat)
+
+        aNeighbors = [
+            self._coefficient_func(self.A_AVG[i], self.A_AMP[i], yearFraction)
+            for i in range(latIndex, latIndex + 2)
+        ]
+        bNeighbors = [
+            self._coefficient_func(self.B_AVG[i], self.B_AMP[i], yearFraction)
+            for i in range(latIndex, latIndex + 2)
+        ]
+        cNeighbors = [
+            self._coefficient_func(self.C_AVG[i], self.C_AMP[i], yearFraction)
+            for i in range(latIndex, latIndex + 2)
+        ]
+
+        # now time to interpolate between them
+        latNeighbors = self.LAT[latIndex : latIndex + 2]
+
+        a = self._interp(np.abs(lat), latNeighbors, aNeighbors)
+        b = self._interp(np.abs(lat), latNeighbors, bNeighbors)
+        c = self._interp(np.abs(lat), latNeighbors, cNeighbors)
+
+        # the base mapping function
+        baseMap = self._herring_map(alt, a, b, c)
+
+        # now add in the mapping correction based on height
+        fcorrection = self._herring_map(alt, self.A_HT, self.B_HT, self.C_HT)
+
+        return baseMap + (1 / np.sin(alt) - fcorrection) * H.to(u.km).value
+
+    def wet_map(self, alt, lat):
+        """this is very similar to the normal mapping function except it uses different coefficients
+        and i believe that there isn't an explicit height correction to the mapping function,
+        but i'll double check the literature.
+        """
+
+        latIndex = self._find_latitude_index(lat)  # lattitude dependent
+
+        aNeighbors = self.AW[latIndex : latIndex + 2]
+        bNeighbors = self.BW[latIndex : latIndex + 2]
+        cNeighbors = self.CW[latIndex : latIndex + 2]
+
+        latNeighbors = self.LAT[latIndex : latIndex + 2]
+
+        a = self._interp(np.abs(lat), latNeighbors, aNeighbors)
+        b = self._interp(np.abs(lat), latNeighbors, bNeighbors)
+        c = self._interp(np.abs(lat), latNeighbors, cNeighbors)
+
+        return self._herring_map(alt, a, b, c)
+
+    @staticmethod
+    def _interp(x, xn, yn):
```

Comment:
Can you not using one of the numpy interpolation methods rather than implementing your own?

Reply:
Ideally I would, but I'm having trouble getting the numpy/scipy interpolation to work with vectorized quantities.  For a TOA observed at 35 degree latitude, I calculate the mapping function at 30 degrees and 45 degrees using the table of values.  This depends on the time of year, so it is a vectorized calculation for each TOA at that observatory.  Then I interpolate between the two results to the correct latitude.  And that's where I get the issues, with numpy not recognizing the vector structure properly and crashing.  I can continue trying to get it to work if it's necessary to use the built-in library for this calculation.